Mangohud failed to build with -Dwith_xnvctrl=disabled and -Dwith_nvml=enabled

Since commit cbf531e6c83a6eff928f80b7399a06dfb1484593

There are error when compiling mangohud with option `-Dwith_xnvctrl=disabled`and `-Dwith_nvml=enabled`

    [63/64] Linking target src/libMangoHud.so
    FAILED: src/libMangoHud.so 
    c++  -o src/libMangoHud.so src/libMangoHud.a.p/mesa_util_os_socket.c.o src/libMangoHud.a.p/mesa_util_os_time.c.o src/libMangoHud.a.p/hud_elements.cpp.o src/libMangoHud.a.p/overlay.cpp.o src/libMangoHud.a.p/overlay_params.cpp.o src/libMangoHud.a.p/font.cpp.o src/libMangoHud.a.p/keybinds.cpp.o src/libMangoHud.a.p/font_unispace.c.o src/libMangoHud.a.p/logging.cpp.o src/libMangoHud.a.p/config.cpp.o src/libMangoHud.a.p/gpu.cpp.o src/libMangoHud.a.p/vulkan.cpp.o src/libMangoHud.a.p/blacklist.cpp.o src/libMangoHud.a.p/file_utils.cpp.o src/libMangoHud.a.p/intel.cpp.o src/libMangoHud.a.p/cpu.cpp.o src/libMangoHud.a.p/memory.cpp.o src/libMangoHud.a.p/iostats.cpp.o src/libMangoHud.a.p/notify.cpp.o src/libMangoHud.a.p/elfhacks.cpp.o src/libMangoHud.a.p/real_dlsym.cpp.o src/libMangoHud.a.p/pci_ids.cpp.o src/libMangoHud.a.p/battery.cpp.o src/libMangoHud.a.p/control.cpp.o src/libMangoHud.a.p/gamepad.cpp.o src/libMangoHud.a.p/amdgpu.cpp.o src/libMangoHud.a.p/nvml.cpp.o src/libMangoHud.a.p/loaders_loader_nvml.cpp.o src/libMangoHud.a.p/loaders_loader_x11.cpp.o src/libMangoHud.a.p/shared_x11.cpp.o src/libMangoHud.a.p/dbus.cpp.o src/libMangoHud.a.p/loaders_loader_dbus.cpp.o src/libMangoHud.a.p/gl_glad.c.o src/libMangoHud.a.p/gl_gl_renderer.cpp.o src/libMangoHud.a.p/gl_gl_hud.cpp.o src/libMangoHud.a.p/gl_inject_egl.cpp.o src/libMangoHud.a.p/loaders_loader_glx.cpp.o src/libMangoHud.a.p/gl_inject_glx.cpp.o src/libMangoHud.a.p/meson-generated_.._.._vk_enum_to_str.c.o -Wl,--as-needed -Wl,--no-undefined -Wl,-O1 -shared -fPIC -Wl,--start-group -Wl,-soname,libMangoHud.so src/libMangoHud.a subprojects/imgui-1.81/libimgui.a subprojects/spdlog-1.8.5/src/libspdlog.a -Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,--exclude-libs,ALL -lGL -Wl,--end-group -Wl,--version-script,/mnt/WD4TBBTRFS/billli11/git/Mangohud/src/mangohud.version -pthread
    /usr/bin/ld: src/libMangoHud.a.p/gpu.cpp.o: in function `getNvidiaGpuInfo(overlay_params const&)':
    gpu.cpp:(.text+0x2bc): undefined reference to `checkXNVCtrl()'
    /usr/bin/ld: gpu.cpp:(.text+0x311): undefined reference to `getNvctrlFanSpeed()'
    collect2: error: ld returned 1 exit status
    ninja: build stopped: subcommand failed.

Reverting the commit fix the problem.

--- a/src/gpu.cpp
+++ b/src/gpu.cpp
@@ -5,7 +5,9 @@
 #include <thread>
 #include <cstring>
 #include <spdlog/spdlog.h>
+#ifdef HAVE_XNVCTRL
 #include "nvctrl.h"
+#endif
 #include "timing.hpp"
 #ifdef HAVE_NVML
 #include "nvidia_info.h"

--- a/src/gpu.cpp
+++ b/src/gpu.cpp
@@ -51,9 +51,11 @@ void getNvidiaGpuInfo(const struct overlay_params& params){
             gpu_info.is_power_throttled = (nvml_throttle_reasons & 0x000000000000008CLL) != 0;
             gpu_info.is_other_throttled = (nvml_throttle_reasons & 0x0000000000000112LL) != 0;
         }
-        static bool nvctrl_available = checkXNVCtrl();
-        if (nvctrl_available)
-            gpu_info.fan_speed = getNvctrlFanSpeed();
+        #ifdef HAVE_XNVCTRL
+            static bool nvctrl_available = checkXNVCtrl();
+            if (nvctrl_available)
+                gpu_info.fan_speed = getNvctrlFanSpeed();
+        #endif
 
         return;
     }

